cql3/Cql.g

```diff
@@ -1467,7 +1467,11 @@ collectionLiteral returns [shared_ptr<cql3::term::raw> value]
 	@init{ std::vector<shared_ptr<cql3::term::raw>> l; }
     : '['
           ( t1=term { l.push_back(t1); } ( ',' tn=term { l.push_back(tn); } )* )?
-      ']' { $value = ::make_shared<cql3::lists::literal>(std::move(l)); }
+      ']' { $value = cql3::expr::as_term_raw(
+                        cql3::expr::collection_constructor{
```

Comment:
Again, I'm curious why we don't want to just call it cql3::expr::collection() instead of collection_constructor?

Reply:
It's not the only way to get a collection (selecting a collection column is another). Also, it explains that we're constructing a collection from more primitive elements.

Reply:
Updated the changelog.